src/System.Memory/src/System/ReadOnlySpan.Portable.cs

```diff
@@ -193,6 +193,11 @@ public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
+                object obj = Unsafe.As<Pinnable<T>, object>(ref Unsafe.AsRef(_pinnable));
+                if (obj is string str && _byteOffset == MemoryExtensions.StringAdjustment && _length == str.Length)
```

Comment:
As in https://github.com/dotnet/corefx/pull/27170/, how about doing the _byteOffset check before doing any of the casting / type checking?

Reply:
Will do.